src/app.rs

```diff
@@ -455,3 +409,48 @@ async fn sources_finished(mutex: Arc<Mutex<TopologyController>>) {
         }
     }
 }
+
+fn get_log_levels(default: &str) -> String {
+    std::env::var("VECTOR_LOG")
+        .or_else(|_| {
+            warn!(message = "Use of $LOG is deprecated. Please use $VECTOR_LOG instead.");
+            std::env::var("LOG")
+        })
+        .unwrap_or_else(|_| match default {
+            "off" => "off".to_owned(),
+            level => [
+                format!("vector={}", level),
+                format!("codec={}", level),
+                format!("vrl={}", level),
+                format!("file_source={}", level),
+                "tower_limit=trace".to_owned(),
+                format!("rdkafka={}", level),
+                format!("buffers={}", level),
+                format!("lapin={}", level),
+                format!("kube={}", level),
+            ]
+            .join(","),
+        })
+}
+
+pub fn build_runtime(threads: Option<usize>, thread_name: &str) -> Result<Runtime, ExitCode> {
+    let mut rt_builder = runtime::Builder::new_multi_thread();
+    rt_builder.enable_all().thread_name(thread_name);
+
+    if let Some(threads) = threads {
+        if threads < 1 {
+            #[allow(clippy::print_stderr)]
+            {
+                eprintln!("The `threads` argument must be greater or equal to 1.");
```

Comment:
`error!()` isn't available yet because the subscriber won't have been created yet?

Reply:
I _think_ so, and because we immediately exit, the error will go nowhere.